src/Data/Ord.purs

```diff
@@ -214,10 +214,12 @@ between low hi x
 abs :: forall a. Ord a => Ring a => a -> a
 abs x = if x >= zero then x else negate x
 
--- | The sign function; always evaluates to either `one` or `negate one`. For
--- | any `x`, we should have `signum x * abs x == x`.
+-- | The sign function; always evaluates to `negate one`, `zero` or `one`.
 signum :: forall a. Ord a => Ring a => a -> a
-signum x = if x >= zero then one else negate one
+signum x =
+  if x < zero then negate one
+  else if x > zero then one
+  else zero
```

Comment:
```suggestion
  else x
```
Returning `x` rather than `zero` ought to match the behaviour of `Math.sign` for `Number`.

Reply:
To clarify, will `signum -0.0` will return -1 whereas `signum 0.0` will return `zero`?

Reply:
I've merged this change, but just wanted to clarify.

Reply:
I think @hdgarrood was referring to the value of `Math.sign(-0.0)` in JS being `-0`.

Reply:
Yeah, we want `signum (-0) == (-0)` and `signum 0 = 0`. Possibly worth adding a test so that this is a bit clearer?

Reply:
Added.